### PR TITLE
ListChunkManager: clean up and fixes

### PR DESCRIPTION
**Pull request by KCodeYT**

### CODE REVIEW

**Comment by joserobjr** [comment]
Hello, thank you for the PR. Can you provide more details on which issue the PR is fixing?

**Comment by KCodeYT** [comment]
Its fixing the issue that getBlockDataAt is returning the block id if the layer is not given

**Comment by KCodeYT** [comment]
And the issue that setBlockIdAt is placing the block id from the parent chunkmanager or itself (if the block is given) and not the block id from the method arguments

File: src/main/java/cn/nukkit/level/ListChunkManager.java [hunk 1]
```diff
@@ -24,99 +24,88 @@ public ListChunkManager(ChunkManager parent) {
...
-        setBlockIdAt(x, y, z, 0, id);
+        this.setBlockIdAt(x, y, z, 0, id);
     }
 
     @Override
     public void setBlockIdAt(int x, int y, int z, int layer, int id) {
-        Optional<Block> optionalBlock = this.blocks.stream().filter(block -> block.getFloorX() == x && block.getFloorY() == y && block.getFloorZ() == z && block.layer == layer).findAny();
-        Block block = optionalBlock.orElse(Block.get(this.getBlockIdAt(x, y, z, layer), this.getBlockDataAt(x, y, z, layer), new Position(x, y, z), layer));
-        this.blocks.remove(block);
-        this.blocks.add(Block.get(this.getBlockIdAt(x, y, z, layer), this.getBlockDataAt(x, y, z, layer), new Position(x, y, z), layer));
```
Comment (KCodeYT) [inline src/main/java/cn/nukkit/level/ListChunkManager.java:65]: First issue

File: src/main/java/cn/nukkit/level/ListChunkManager.java [hunk 2]
```diff
@@ -24,99 +24,88 @@ public ListChunkManager(ChunkManager parent) {
...
     @Override
     public int getBlockDataAt(int x, int y, int z, int layer) {
-        Optional<Block> optionalBlock = this.blocks.stream().filter(block -> block.getFloorX() == x && block.getFloorY() == y && block.getFloorZ() == z && block.layer == layer).findAny();
-        return optionalBlock.map(Block::getDamage).orElseGet(() -> this.parent.getBlockDataAt(x, y, z, layer));
+        return this.findBlockAt(x, y, z, layer).map(Block::getDamage).orElseGet(() -> this.parent.getBlockDataAt(x, y, z, layer));
     }
 
     @Override
     public void setBlockDataAt(int x, int y, int z, int data) {
-        setBlockIdAt(x, y, z, 0, data);
```
Comment (KCodeYT) [inline src/main/java/cn/nukkit/level/ListChunkManager.java:110]: Second issue

**Pull request closed without merging**